src/ORM/DataExtension.php

```diff
@@ -13,7 +13,8 @@
 /**
  * An extension that adds additional functionality to a {@link DataObject}.
  *
- * @property DataObject $owner
+ * @template T of DataObject
+ * @extends Extension<T|static>
```

Comment:
On another note, probably wouldn't hurt to go through and find all the subclasses of `Extension` in framework and apply this as part of this PR if you feel like it

Reply:
Added an example implementation on `Hierachy` :)